pkg/workloadmeta/collectors/internal/docker/docker.go

```diff
@@ -354,6 +355,25 @@ func extractImage(ctx context.Context, container types.ContainerJSON, resolve re
 	image.ShortName = shortName
 	image.Tag = tag
 
+	// Get imageID from running container with mathing ID
+	var imageID string
+	filter, err := containers.GetPauseContainerFilter()
+	if err != nil {
+		log.Warnf("Can't get pause container filter, no filtering will be applied: %v", err)
+	}
+	running_containers, err := resolveContainersList(ctx, types.ContainerListOptions{}, filter)
+	if err != nil {
+		log.Debugf("cannot get running containers: %v", err)
+	}
+
+	for _, c := range running_containers {
+		if c.ID == container.ContainerJSONBase.ID {
+			imageID = c.ImageID
+		}
+	}
+	// Add imageID to image
+	image.ID = imageID
+
```

Comment:
Why can’t we simply read the `Image` field from [`container types.ContainerJSON`](https://github.com/moby/moby/blob/v23.0.0-rc.1/api/types/types.go#L457) ?

```suggestion
	image.ID = container.Image
```

Reply:
Yes that's better. I thought the Image field was just the image's name and not the actual Id.